cores/esp8266/Updater.cpp

```diff
@@ -263,8 +272,10 @@ bool UpdaterClass::end(bool evenIfRemaining){
 #endif
     if (!_verify->verify(_hash, (void *)sig, sigLen)) {
       _setError(UPDATE_ERROR_SIGN);
+      free(sig);
```

Comment:
You need to call `_reset()` as well for deallocating the buffer and eventually turning off the LED.

Reply:
Seems like that's a requisite for all `return false` paths in this `::end`, too.

Reply:
Just added in the latest push.